controllers/apps/v1beta3/emqx_handler.go

```diff
@@ -312,24 +286,14 @@ func generateStatefulSetDef(instance appsv1beta3.Emqx) *appsv1.StatefulSet {
 					ImagePullSecrets: instance.GetImagePullSecrets(),
 					SecurityContext:  instance.GetSecurityContext(),
 					InitContainers:   instance.GetInitContainers(),
-					Containers: []corev1.Container{
-						{
-							Name:            "emqx",
-							Image:           instance.GetImage(),
-							ImagePullPolicy: instance.GetImagePullPolicy(),
-							Resources:       instance.GetResource(),
-							Env:             mergeEnvAndConfig(instance),
-							Args:            instance.GetArgs(),
-							ReadinessProbe:  instance.GetReadinessProbe(),
-							LivenessProbe:   instance.GetLivenessProbe(),
-							StartupProbe:    instance.GetStartupProbe(),
-						},
-					},
 				},
 			},
 		},
 	}
-
+	emqxContainer := generateEmqxContainer(instance)
+	sts.Spec.Template.Spec.Containers = append(sts.Spec.Template.Spec.Containers, emqxContainer)
+	//add other container
+	sts.Spec.Template.Spec.Containers = append(sts.Spec.Template.Spec.Containers, instance.GetExtraContainers()...)
```

Comment:
Maybe reloader container can insert sts in this?  And update sts manager containers annotation in this